srtcore/channel.cpp

```diff
@@ -407,24 +420,47 @@ int CChannel::recvfrom(sockaddr* addr, CPacket& packet) const
     int msg_flags = 0;
 #endif
 
-    // These logs are theoretically errors, but this isn't anything problematic
-    // for the application, and in certain conditions they can be spit out very
-    // often and therefore influence the processing time.
+    // Note that there are exactly four groups of possible errors
+    // reported by recvmsg():
+
+    // 1. Temporary error, can't get the data, but you can try again.
+    // Codes: EAGAIN/EWOULDBLOCK, EINTR
+    // Return: RST_AGAIN.
+    //
+    // 2. Problems that should never happen due to unused configurations.
+    // Codes: ECONNREFUSED, ENOTCONN
+    // Return: RST_ERROR, just formally treat this as IPE.
+    //
+    // 3. Unexpected runtime errors:
+    // Codes: EINVAL, EFAULT, ENOMEM, ENOTSOCK
+    // Return: RST_ERROR. Except ENOMEM, this can only be an IPE. ENOMEM
+    // should make the program stop as lacking memory will kill the program anyway soon.
+    //
+    // 4. Expected socket closed in the meantime by another thread.
+    // Codes: EBADF
+    // Return: RST_ERROR. This will simply make the worker thread exit, which is
+    // expected to happen after CChannel::close() is called by another thread.
+
     if ( res == -1 )
     {
-#if ENABLE_LOGGING
         int err = NET_ERROR;
-        if ( err != EAGAIN ) // For EAGAIN, this isn't an error, just a useless call.
+        if ( err == EAGAIN || err == EINTR ) // For EAGAIN, this isn't an error, just a useless call.
```

Comment:
How come is EINTR a retry later. Most probably an application terminating  I may have found why my CLI test app do not terminate as before on Ctrl-C.

Reply:
EINTR is "interrupted system call". It can also happen when an application gets SIGUSR1 sent from some other process because the application was designed to work with this. This signal is theoretically unrelated to SRT, but still it's a signal and it will interrupt a system call, like every signal does. This is generally a problem for libraries, and a library must be as transparent as possible for signals. Not sure if EINTR is even possible in nonblocking mode, but at least I can imagine such a situation.

Reply:
EINTR is fatal for most normal applications and library should have means of indicating this condition to the main application code. Please do not treat it as "try again".

Reply:
Please do not discuss such details here.

Reply:
Hello @ethouris , I don't think Jean and me are looking for a story but rather for explanation why we should be masking EINTR in this particular case. Also I believe Jean thinks this is already breaking his app so if this change is to blame for CTRL-C not working it is a bug.